posts/2024-08-20-liberty-olo-rh-sso.adoc

```diff
@@ -0,0 +1,338 @@
+
+. Create a realm named `sso-realm`. Use the following URL to access this realm:
++
+`\https://keycloak-rh-sso.apps.<cluster-name>/auth/admin/master/console/#/realms/sso-realm`
+
+. Create a non-admin user for this realm.
++
+Create a user called `testuser1` by selecting **Manage** > **Users** > **Add user**. Specify `testuser1` for **Username** and click **Save**. We can use this user to test the social login when RH-SSO is used as an OIDC provider.
++
+[.img_border_light]
+image::img/blog/rh_create_testuser1.png[Create testuser1,width=50%,align="center"]
++
+. Select the **Credentials** tab and specify `testpasswd1` as the password for the user you created.
+. Change the **Temporary** option from `ON` to `Off` and click **Reset Password**.
+. On the confirmation dialog, select **Change Password**.
+. Go to the **Role Mappings** tab.
+. Click **Client Roles** menu and select **realm-management**. After you make this selection, boxes such as **Available Roles** appear.
+. Under **Available Roles**, search for **view-realm**, select it, and click **Add selected**. After the role is selected, it appears under `Assigned Roles` and `Effective Roles`.
++
+__Note: Selecting the role is just a basic requirement to allow the user to log in to the console on RH-SSO. If no role is assigned, the user sees a Forbidden error message in the browser.__
++
+[.img_border_light]
+image::img/blog/rh_testuser1_roles.png[Roles for testuser1,width=50%,align="center"]
+
+. Test the user you created.
++
+Go to the `\https://keycloak-rh-sso.apps.<cluster-name>/auth/admin/Sso-realm/console/` URL and log in by using the `testuser1`/`testpasswd1` username you created in the previous steps. After you log in, in the **General** section, you can see the endpoints.
+. Click the link for the **OpenID Endpoint Configuration**, which points you to the following URL:
++
+`\https://keycloak-rh-sso.apps.<cluster-name>/auth/realms/sso-realm/.well-known/openid-configuration`.
++
+You need this URL as the `discoveryEndpoint` for the client registration later on.
+
+[#createSecret]
+== Create the olapp-sso secret
+
+Next, we create a secret for the Open Liberty application. The key name for this secret is in the `<app-name>-olapp-sso` format. You must use the same `<app-name>` as the one you use to deploy the application with OLO. For example, in the application that is used in this post, the `<app-name>` from the YAML file is `guide-social-media-login`, so the secret name is `guide-social-media-login-olapp-sso`.
+
+. Create a project in your cluster called `gsm-test`.
```

Comment:
> Create a project in your cluster called `gsm-test`.

How? is this documented on the RH side somewhere that we can point to?

Reply:
https://docs.openshift.com/container-platform/4.16/applications/projects/working-with-projects.html documents how to create a project.